manager/orchestrator/replicated/services.go

```diff
@@ -124,7 +124,12 @@ func (r *Orchestrator) reconcile(ctx context.Context, service *api.Service) {
 		// Preferentially remove tasks on the nodes that have the most
 		// copies of this service, to leave a more balanced result.
 
-		// First sort tasks such that tasks which are currently running
+		// First sort tasks in an ascending order of slot number. This will
+		// make the task termination order deterministic.
+
+		sort.Sort(slotsBySlotNumber(slotsSlice))
```

Comment:
golang's `Sort` function is not a stable sort, so I don't think this accomplishes anything. The same slice is immediately re-sorted afterwards using `slotsByRunningState`. I think the right way to do this is to change `slotsByRunningState` so that slot number is used as a tie-breaker.